core/src/main/java/com/linecorp/armeria/server/cors/CorsServiceBuilder.java

```diff
@@ -68,48 +62,49 @@ public static CorsServiceBuilder forOrigin(final String origin) {
      */
     public static CorsServiceBuilder forOrigins(final String... origins) {
         requireNonNull(origins, "origins");
-        for (int i = 0; i < origins.length; i++) {
-            if (origins[i] == null) {
-                throw new NullPointerException("origins[" + i + ']');
-            }
-        }
         return new CorsServiceBuilder(origins);
     }
 
-    final Set<String> origins;
     final boolean anyOriginSupported;
-    boolean nullOriginAllowed;
-    boolean credentialsAllowed;
+    final ChainedCorsPolicyBuilder defaultPolicyBuilder;
```

Comment:
I'm curious if this class can be simplified by extending `AbstractCorsServiceBuilder` instead of delegating all builder methods. Thoughts?